src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,873 @@
+						{
+							var firstChildNode = changingChildrenNode.Children[0];
+							UpdateSelectionStateOfAncestors(firstChildNode);
+						}
+
+						var selectedNodes = get_self<SelectedTreeNodeVector>(m_selectedNodes);
+						for (unsigned int i = 0; i < selectedNodes.Size(); i++)
+						{
+							var selectNode = selectedNodes.GetAt(i);
+							var ancestorNode = selectNode.Parent();
+							while (ancestorNode && ancestorNode.Parent())
+							{
+								ancestorNode = ancestorNode.Parent();
+							}
+
+							if (ancestorNode != m_originNode.get())
+							{
+								selectedNodes.RemoveAtCore(i);
+								m_selectedNodeChildrenChangedEventTokenVector.erase(m_selectedNodeChildrenChangedEventTokenVector.begin() + i);
+							}
+						}
+						break;
+					}
+
+			}
+		}
+
+		private void TreeViewNodePropertyChanged(TreeViewNode sender, DependencyPropertyChangedEventArgs args)
+		{
+			DependencyProperty property = args.Property;
+			if (property == TreeViewNode.IsExpandedProperty)
+			{
+				TreeViewNodeIsExpandedPropertyChanged(sender, args);
+			}
+			else if (property == TreeViewNode.HasChildrenProperty)
+			{
+				TreeViewNodeHasChildrenPropertyChanged(sender, args);
+			}
+		}
+
+		private void TreeViewNodeIsExpandedPropertyChanged(TreeViewNode sender, DependencyPropertyChangedEventArgs args)
+		{
+			var targetNode = (TreeViewNode)sender;
+			if (targetNode.IsExpanded)
+			{
+				if (targetNode.Children.Count != 0)
+				{
+					int openedDescendantOffset = 0;
+					IndexOfNode(targetNode, out var index);
+					index = index + 1;
+					for (var i = 0; i < targetNode.Children.Count; i++)
+					{
+						TreeViewNode childNode = null;
+						childNode = (TreeViewNode)targetNode.Children[i];
+						AddNodeToView(childNode, index + i + openedDescendantOffset);
+						openedDescendantOffset = AddNodeDescendantsToView(childNode, index + i, openedDescendantOffset);
+					}
+				}
+
+				//Notify TreeView that a node is being expanded.
+				m_nodeExpandingEventSource(targetNode, null);
+			}
+			else
+			{
+				for (var i = 0; i < targetNode.Children.Count; i++)
+				{
+					TreeViewNode childNode = null;
+					childNode = (TreeViewNode)targetNode.Children[i];
+					RemoveNodeAndDescendantsFromView(childNode);
+				}
+
+				//Notife TreeView that a node is being collapsed
+				m_nodeCollapsedEventSource(targetNode, null);
+			}
+		}
+
+		private void TreeViewNodeHasChildrenPropertyChanged(TreeViewNode sender, DependencyPropertyChangedEventArgs args)
+		{
+			if (m_TreeViewList != null)
+			{
+				var targetNode = sender;
+				var container = m_TreeViewList.ContainerFromNode(targetNode);
+				if (container != null)
+				{
+					TreeViewItem targetItem = (TreeViewItem)container;
+					targetItem.GlyphOpacity = targetNode.HasChildren ? 1.0 : 0.0);
+				}
+			}
+		}
+
+		public bool IsContentMode { get => m_isContentMode; set => m_isContentMode = value; }
+
+		//void ViewModel.ClearEventTokenVectors()
+		//{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)